terraform/bin/manage-stack.sh

```diff
@@ -493,14 +619,105 @@ function do_stack {
     #else: Nothing to delete 
     fi
     extra_verbose "About to [${action}_stack ${stack_args[@]}]"
-    pushd $(pwd) 1> /dev/null
+    pushd "${work_dir}" 1> /dev/null
     "${action}_stack" ${stack_args[@]} 
     status=$?
     popd 1> /dev/null
     return $status
 }
 
 
+
+function instance_passes_healthcheck {
+    local resource="$1" status
+    # Find the instance
+    local instanceID=$(ec2 describe-instances \
+            --filter "Name=tag:System,Values=datagov" \
+                     "Name=tag:Stack,Values=${STACK_NAME}" \
+                     "Name=tag:Branch,Values=${BRANCH_NAME}" \
+                     "Name=tag:Resource,Values=${resource}" \
+                     "Name=instance-state-name,Values=running" \
+            --query "Reservations[].Instances[].{Id:InstanceId}" \
+            --output text)
+    if [ "${instanceID}" == "" ]; then
+        debug "Resource ${resource} not found"
+        return 1
+    fi
+    # Check the health-check status
+    status=$(ec2 describe-instance-status \
+        --instance-ids "${instanceID}" \
+        --filter "Name=instance-state-name,Values=running" \
+                 "Name=instance-status.reachability,Values=passed" \
+                 "Name=system-status.reachability,Values=passed" \
+        --query "InstanceStatuses[].{Id:InstanceId}" \
+        --output text)
+    if [ "${status}" == "" ]; then
+        debug "Resource ${resource} health checks not passed"
+        return 1
+    fi
+    debug "Resource ${resource} health checks passed"
+    return 0
+}
+
+function all_resources_completed {
```

Comment:
so effectively this allows the terraform script to check that all resources are completed including any health checks? This is what will be used to tell ansible to proceed in its execution?